# Лабораторная работа 3.2. Домашнее задание

### Задание №1

Создайте два массива: в первом должны быть четные числа от 2 до 12 включительно, а в другом числа 7, 11, 15, 18, 23, 29.

$1.$ Сложите массивы и возведите элементы получившегося массива в квадрат:

In [4]:
import numpy as np

arr1 = np.arange(2, 13, 2)
arr2 = np.array([7, 11, 15, 18, 23, 29])

print((arr1 + arr2) ** 2)

[  81  225  441  676 1089 1681]


$2.$ Выведите все элементы из первого массива, индексы которых соответствуют индексам тех элементов второго массива, которые больше 12 и дают остаток 3 при делении на 5.

In [6]:
print(arr1[(arr2 > 12) & (arr2 % 5 == 3)])

[ 8 10]


*3.* Проверьте условие "Элементы первого массива делятся на 4, элементы второго массива меньше 14". (Подсказка: в результате должен получиться массив с True и False)

In [7]:
print("First condition: ", arr1 % 4 == 0)
print("Second condition: ", arr2 < 14)

First condition:  [False  True False  True False  True]
Second condition:  [ True  True False False False False]


### Задание №2

* Найдите интересный для вас датасет. Например, можно выбрать датасет тут. http://data.un.org/Explorer.aspx (выбираете датасет, жмете на view data, потом download, выбирайте csv формат)
* Рассчитайте подходящие описательные статистики для признаков объектов в выбранном датасете
* Проанализируйте и прокомментируйте содержательно получившиеся результаты
* Все комментарии оформляйте строго в ячейках формата markdown
    

In [12]:
import csv
import numpy as np

with open('Customers.csv', 'r', newline='') as csvf:
    data = csv.reader(csvf, delimiter=',')
    age = []
    income = []
    for row in data:
        if row[0] == "CustomerID":
            continue
        age.append(int(row[2]))   
        income.append(int(row[3]))
       
        
age_arr = np.array(age)
income_arr = np.array(income)

In [16]:
print(f"average age value: {round(np.mean(age_arr),3)}")
print(f"average value of income: {round(np.mean(income_arr),3)}")

print(f"\nmaximum age in the sample: {np.max(age_arr)}")
print(f"maximum income in the sample: {np.max(income_arr)}")

print(f"\nminimum age in the sample: {np.min(age_arr)}")
print(f"minimum income in the sample: {np.min(income_arr)}")

print(f"\naverage deviation for age: {round(np.std(age_arr),3)}")
print(f"average deviation for income: {round(np.std(income_arr),3)}")

average age value: 48.96
average value of income: 110731.822

maximum age in the sample: 99
maximum income in the sample: 189974

minimum age in the sample: 0
minimum income in the sample: 0

average deviation for age: 28.423
average deviation for income: 45728.1


*1. the age of the survey participants is from 0 to 99*

*2. average devitation for age is 28.423 for income it is 45728.1*